### ISSUE TITLE
JavaScript: Object functions containing function calls are not recognized 

### ISSUE THREAD

**Issue by gpavlov2016** [universal-ctags/ctags#1933]
Example:
```
testobj = {
  func1: () => {
    console.log('test1');
  },
  func2: () => {
    console.log('test2');
  }
};
```
In this case func1 is recognized but func2 is not.
Interesting that if the call inside func1 is removed then both func1 and func2 are recognized:
```
testobj = {
  func1: () => {
  },
  func2: () => {
    console.log('test2');
  }
};

```

**Comment by masatake** [comment]
What language are you talking about?

**Comment by gpavlov2016** [comment]
JavaScript ES6

**Comment by gpavlov2016** [comment]
> What language are you talking about?

JavaScript ES6

**Comment by masatake** [comment]
@b4n, can I ask you to look at this?

**Comment by masatake** [comment]
```
diff --git a/parsers/jscript.c b/parsers/jscript.c
index 458b5cb1..bb75637d 100644
--- a/parsers/jscript.c
+++ b/parsers/jscript.c
Following patch makes ctags capture func2. Howerver, I cannot convince myself this is the correct fix.
Let's wait how @b4n says.

@@ -1173,13 +1207,13 @@ static void addToScope (tokenInfo* const token, const vString* const extra)
  */
 
 static bool findCmdTerm (tokenInfo *const token, bool include_newlines,
-                            bool include_commas)
+                            bool include_commas, bool ignore_semicolon)
 {
 	/*
 	 * Read until we find either a semicolon or closing brace.
 	 * Any nested braces will be handled within.
 	 */
-	while (! isType (token, TOKEN_SEMICOLON) &&
+	while (! ((! ignore_semicolon) && isType (token, TOKEN_SEMICOLON)) &&
 		   ! isType (token, TOKEN_CLOSE_CURLY) &&
 		   ! (include_commas && isType (token, TOKEN_COMMA)) &&
 		   ! isType (token, TOKEN_EOF))
@@ -1381,7 +1415,7 @@ static bool parseIf (tokenInfo *const token)
 	{
 		/* The next token should only be read if this statement had its own
 		 * terminator */
-		read_next_token = findCmdTerm (token, true, false);
+		read_next_token = findCmdTerm (token, true, false, false);
 	}
 	return read_next_token;
 }
@@ -1440,7 +1474,7 @@ static void parseFunction (tokenInfo *const token)
 			makeFunctionTagCommon (name, signature, is_generator, is_anonymous);
 	}
 
-	findCmdTerm (token, false, false);
+	findCmdTerm (token, false, false, false);
 
  cleanUp:
 	vStringDelete (signature);
@@ -1687,7 +1721,7 @@ static bool parseMethods (tokenInfo *const token, const tokenInfo *const class,
 
 	JSCRIPT_DEBUG_PRINT("Finished parsing methods");
 
-	findCmdTerm (token, false, false);
+	findCmdTerm (token, false, false, true);
 
 cleanUp:
 	vStringCopy (token->scope, saveScope);
@@ -1989,7 +2023,7 @@ nextVar:
 							/*
 							 * Find to the end of the statement
 							 */
-							findCmdTerm (token, false, false);
+							findCmdTerm (token, false, false, false);
 							token->ignoreTag = false;
 							is_terminated = true;
 							goto cleanUp;
@@ -2330,7 +2364,7 @@ nextVar:
 		 *	   return 1;
 		 * }
 		 */
-		is_terminated = findCmdTerm (token, true, true);
+		is_terminated = findCmdTerm (token, true, true, false);
 		/* if we're at a comma, try and read a second var */
 		if (isType (token, TOKEN_COMMA))
 		{
```

**Comment by b4n** [comment]
I'll look into it, but it might take a little while.

**Comment by thisguychris** [comment]
I've tried @masatake patch, I still couldn't get it to work.
`func2` though can be successfully parsed if you add a variable declaration on the object:

```javascript
const testobj = {
  func1: () => {
    console.log('test1');
  },
  func2: () => {
    console.log('test2');
  }
};
```

**Comment by jafl** [comment]
@masatake Please close this, because it was fixed by #3357

**Comment by masatake** [comment]
@jafl, thank you.
